syntax = "proto3";


option objc_class_prefix = "Listing";

package listing;

service Listing {
  rpc GetStocksInExchange(ExchangeRequest) returns(stream StockListing) {}
  rpc GetStockListing(StockRequest) returns(StockListing) {}
}

message Empty {}

message ExchangeRequest {
    string code = 1;
}

message StockRequest {
    string symbol = 1;
}

message StockListing{
    int32 conId = 1;
    string symbol = 2;
    string currency = 3;
    string exchange = 4;
}